src/nvim/garray.c

```diff
@@ -178,7 +178,7 @@ char_u *ga_concat_strings_sep(const garray_T *gap, const char *sep)
 /// @param gap
 ///
 /// @returns the concatenated strings
-char_u* ga_concat_strings(const garray_T *gap) FUNC_ATTR_NONNULL_RET
+char_u * ga_concat_strings(const garray_T *gap) FUNC_ATTR_NONNULL_RET
```

Comment:
? is it doing this for all functions? not sure we want that. doesn't matter for now i guess.

Reply:
Ew, -1 for this.

Reply:
Added a rules that removes space between a start and functions name.

Reply:
Personally, I prefer to put the `*` next to the identifier rather than the type. This avoids confusion in cases like

```c
int* p, x;
```

The above implies that both `p` and `x` are pointers to int `int *` which is not the case: only `p` is and `x` is just an `int` (yet another reason to avoid multiple declarations on one line).

But seeing as there's no ambiguity in a function declaration this is fine too.

Reply:
I don't get it. The change I did is the same one as the one you advocate for?

Reply:
No, that change would be `char_u *ga_concat_strings` (which isn't as obviously better as `int *p`, though).

Reply:
Sorry, I updated my comment to make it less confusing. I am advocating for `char_u *foo()` rather than `char_u* foo()`, but as I said it matters less for function declarations than it does for variable declarations.

Reply:
> Personally, I prefer to put the `*` next to the identifier rather than the type:

this is the current style guideline. I think uncrustify just needed a tweak for function return types, which was made in this PR.

Reply:
You're right, I hadn't looked at the diff :) My mistake.